[ALL] Completamento glossario (#137)
Argo-swe/Docs: sources/documents/2_RTB/Glossario/Lettere/b.tex

\section{B}

\vspace{2em}

> \subsection*{Backend}

Reviewer: vanno bene entrambe le forme alla fine. L'ho segnato nel Piano di Qualifica.

%Una base di dati è una raccolta organizzata di dati strutturati, generalmente memorizzata e accessibile elettronicamente da un sistema informatico. Le basi di dati sono progettate per consentire una gestione efficiente dei dati, facilitando operazioni di inserimento, aggiornamento, eliminazione e interrogazione. Utilizzano sistemi di gestione delle basi di dati (DBMS) per garantire integrità, sicurezza e disponibilità dei dati. Le basi di dati possono essere di diversi tipi, tra cui relazionali, NoSQL, gerarchiche e a grafo, ognuna adatta a specifiche esigenze applicative.
\par Vedi \glossario{database}

\vspace{2em}
\subsection*{Benchmark}
TODO

\vspace{2em}
\subsection*{Best practices}
Le best practices, o migliori pratiche, sono metodologie, processi o tecniche riconosciute come standard di riferimento per l'esecuzione di determinate attività o per il raggiungimento di obiettivi specifici. Sono sviluppate attraverso l'esperienza, l'analisi dei risultati e il confronto con altre organizzazioni o professionisti del settore e sono considerate efficaci e efficienti nel fornire risultati desiderati.

\vspace{2em}
\subsection*{Bootstrap}
TODO

\vspace{2em}
\subsection*{Branch}

> Un branch rappresenta una linea di sviluppo indipendente. I rami fungono da astrazione per il processo di modifica/stage/commit. Vengono considerati come una directory di lavoro, un'area di staging e una cronologia del progetto completamente nuove. I nuovi commit vengono registrati nella cronologia del ramo corrente, il che si traduce in un fork nella cronologia del progetto.

Reviewer: ```suggestion
Un branch rappresenta una linea di sviluppo indipendente. I rami fungono da astrazione per il processo di modifica/stage/commit. Vengono considerati come una directory di lavoro, un'area di staging e una cronologia del progetto completamente nuove. I commit vengono registrati nella cronologia del ramo corrente, il che si traduce in un fork nella cronologia del progetto.
```

\vspace{2em}
\subsection*{Bug}

> Un bub è un'anomalia che porta al malfunzionamento di un software, per esempio producendo un risultato inatteso o errato, tipicamente dovuto a un errore nella scrittura del codice sorgente di un programma. Un bug, in sostanza, è un difetto software di tipo funzionale, nonostante a volte il termine venga usato per indicare anche falle di sicurezza (vulnerabilità) nonché per gravi carenze di prestazione.

Reviewer: ```suggestion
Un bug è un'anomalia che porta al malfunzionamento di un software, per esempio producendo un risultato inatteso o errato, tipicamente dovuto a un errore nella scrittura del codice sorgente di un programma. Un bug, in sostanza, è un difetto software di tipo funzionale, nonostante a volte il termine venga usato per indicare anche falle di sicurezza (vulnerabilità) o per evidenziare inadeguatezze prestazionali.
```

\vspace{2em}
\subsection*{Build}

> Una build è il processo di trasformazione del codice sorgente in un artefatto eseguibile. Talvolta lo stesso termine viene utilizzato non per indicare il processo ma il relativo risultato finale ovvero l'artefatto ottenuto o il corrispondente codice numerico (o alfanumerico) che permette di distinguere tale artefatto da altri ottenuti in precedenza nell'ambito dell'evoluzione dello stesso prodotto software.

Reviewer: ```suggestion
Una build è un processo di trasformazione del codice sorgente in un artefatto eseguibile. Talvolta lo stesso termine viene utilizzato per indicare non tanto il processo quanto il risultato finale, ovvero l'artefatto ottenuto o il corrispondente codice numerico (o alfanumerico). Tale codice permette di distinguere l'artefatto da quelli ottenuti in precedenza nell'ambito dell'evoluzione di un prodotto software.
```
